src/client/client.go

```diff
@@ -42,11 +42,15 @@ type APIClient struct {
 	cancel            func()
 	reportUserMetrics bool
 	metricsPrefix     string
+	streamSemaphores  chan struct{}
```

Comment:
Minor knit, don't think this should be plural since it's just a single semaphore.